frontend/public/components/factory/list-page.jsx

```diff
@@ -205,7 +207,10 @@ export const FireMan_ = connect(null, {filterList})(
           </Link>;
         } else if (createProps.items) {
           createLink = <div className="co-m-primary-action">
-            <Dropdown buttonClassName="btn-primary" id="item-create" title={createButtonText} noSelection={true} items={createProps.items} onChange={(name) => history.push(createProps.createLink(name))} />
+            <Dropdown buttonClassName="btn-primary" id="item-create" title={createButtonText}
+              noSelection={true}
+              items={createProps.items}
+              onChange={(name) => runOrNavigate(createProps.createLink(name))} />
```

Comment:
I would rather have a separate prop for `onClick` instead of overloading `createLink` to mean more than one thing

Reply:
Ok, keeping the `createLink` as it was before.
As it is a dropdown, I think it makes sense to keep using the `onChange` here.

So I am adding new `action` prop to `createProps`, which is a function with the same signature as the `createLink`. If it returns a function for given `item`, then it is fired. Otherwise the `createLink` is used for navigation as before.

@spadgett , @dtaylor113 , can it be like this?

Reply:
> I would rather have a separate prop for onClick instead of overloading createLink to mean more than one thing

@spadgett I'm new here and may be missing some context, 
do you mean to add `onClick` here:
https://github.com/openshift/console/blob/master/frontend/public/components/utils/dropdown.jsx#L53